doc/changelog.md

```diff
@@ -15,6 +15,9 @@
   [(#895)](https://github.com/PennyLaneAI/catalyst/pull/895)
 
 <h3>Bug fixes</h3>
+* Using float32 in callback functions would not crash in compilation phase anymore,
+  but rather raise an assertion error.
```

Comment:
Note that we actually don't raise an assertion error (the type of error raised in Python by the `assert` statement). Looking at your, we raise a `TypeError`.

```suggestion
  but rather raise the appropriate type exception to the user.
```

For development, we are careful to distinguish between assertions and exceptions. Assertions can (in principle) be disabled, and are only meant to verify internal developer assumptions. Anything that can make its way to the user should be a "proper" exception.

Reply:
Yes, you are right. Thanks for catching this. I fixed the changelog